src/ensembl/src/content/app/browser/track-panel/trackPanelActions.ts

```diff
@@ -99,10 +106,62 @@ export const changeTrackPanelModalViewForGenome: ActionCreator<
   );
 };
 
+export const updatePreviouslyViewedObjectsAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = () => (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const activeGenomeId = getBrowserActiveGenomeId(state);
+  const activeEnsObject = getBrowserActiveEnsObject(state);
+  if (!activeGenomeId || !activeEnsObject) {
+    return;
+  }
+  const trackStates = getBrowserTrackStates(state)[activeGenomeId];
+
+  const previouslyViewedObjects = [
+    ...getActiveGenomePreviouslyViewedObjects(state)
+  ];
+
+  const existingIndex = previouslyViewedObjects.findIndex(
+    (previouslyViewedObject) =>
+      previouslyViewedObject.object_id === activeEnsObject.object_id
+  );
+  if (existingIndex === -1) {
+    // IF it is not present, add it to the end
+    previouslyViewedObjects.push({
+      genome_id: activeEnsObject.genome_id,
+      object_id: activeEnsObject.object_id,
+      object_type: activeEnsObject.object_type,
+      label: activeEnsObject.label,
+      trackStates: { ...trackStates }
```

Comment:
why the spread (...trackStates)?

Reply:
Removed it.

Reply:
If we want to have a completely new trackStates object here, you can use something like `cloneDeep` from lodash